src/components/Header.tsx

```diff
@@ -0,0 +1,103 @@
+import { useEffect, useState } from 'react';
+import { Todo } from '../types/Todo';
+import cn from 'classnames';
+import * as todoService from './../api/todos';
+
+type Props = {
+  todos: Todo[];
+  setValueError: (e: string) => void;
+  setTempTodo: React.Dispatch<React.SetStateAction<Todo | null>>;
+  setIsNotificationHidden: (value: boolean) => void;
+  autoHideNotification: () => void;
+  onAdd: (todo: Todo) => void;
+  inputRef: React.RefObject<HTMLInputElement>;
+};
+
+export const Header: React.FC<Props> = ({
+  todos,
+  setValueError,
+  setIsNotificationHidden,
+  autoHideNotification,
+  onAdd,
+  setTempTodo,
+  inputRef,
+}) => {
+  const [valueTitle, setValueTitle] = useState('');
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (valueTitle.trim() === '') {
+      setValueError('Title should not be empty');
+      setIsNotificationHidden(false);
+      autoHideNotification();
+      inputRef.current?.focus();
+
+      return;
+    }
+
+    const trimmedTitle = valueTitle.trim();
+    const newTodo: Todo = {
+      id: 0,
+      title: trimmedTitle,
+      userId: todoService.USER_ID,
+      completed: false,
+    };
+
+    setTempTodo(newTodo);
+
+    try {
+      setIsSubmitting(true);
+
+      const createdTodo = await todoService.createTodo({
+        title: trimmedTitle,
+        userId: newTodo.userId,
+        completed: newTodo.completed,
+      });
+
+      onAdd(createdTodo);
+
+      setValueTitle('');
+      setTempTodo(null);
+      inputRef.current?.focus();
```

Comment:
We can create useEffect in App.tsx which will focus input when there are changes and you won't repeat so many times